clients/fides-js/src/lib/cookie.ts

```diff
@@ -28,18 +28,21 @@ export const CONSENT_COOKIE_NAME = "fides_consent";
 export const CONSENT_COOKIE_MAX_AGE_DAYS = 365;
 
 /**
- * The typescript-cookie default codec has a more conservative strategy in order to
+ * The js-cookie default codec has a more conservative strategy in order to
  * comply with the exact requirements of RFC 6265. For ease of use in external pages,
  * we instead use encode/decodeURIComponent which are available in every browser.
  *
- * See: https://github.com/carhartl/typescript-cookie#encoding
+ * See: https://www.npmjs.com/package/js-cookie#converters
  */
-const CODEC: Types.CookieCodecConfig<string, string> = {
-  decodeName: decodeURIComponent,
-  decodeValue: decodeURIComponent,
-  encodeName: encodeURIComponent,
-  encodeValue: encodeURIComponent,
-};
+
+const cookies: CookiesStatic = Cookies.withConverter({
```

Comment:
This `withConvertor` method allows us to define a custom CODEC for encoding/decoding the name/val of cookies.

Reply:
Since you're doing this statically when the `cookie.ts` file is loaded, this might be why your mock isn't working. I have a feeling this code runs against the real (non-mocked) version, and then afterwards your code registers the spy.

You could try adding some logging or breakpoints to check for that.

Reply:
Yes this was the tricky bit, but @gilluminate was able to figure out the correct way to mock this!